addons/web/static/src/js/views/form/form_controller.js

```diff
@@ -70,6 +70,18 @@ var FormController = BasicController.extend({
             return self._setMode('edit');
         });
     },
+    toEditMode: function() {
+        return this._setMode('edit');
+    },
+    nextPage:function() {
+        return this._nextPage();
+    },
+    previousPage:function() {
+        return this._previousPage();
+    },
+    discardChange: function() {
+        return this.discardChanges();
+    },
```

Comment:
i think that those functions are not necessary. You could directly call the right function in your include in barcodes (a private function can't be called by the outside world, but in this case you are still in the form_controller.
For toEditMode, you can define an inline function as you already did for 'O-CMD.SAVE'